### PR TITLE
cilium-dbg: Add "bpf ipcache delete/update"

### PR DESCRIPTION
**Pull request by jschwinger233**
https://github.com/cilium/cilium/pull/35231 will use two new ipcache cmd to temporarily delete ipcache entries for strict mode encryption test.

Suggested-by: Timo Beckers <timo@isovalent.com>

### CODE REVIEW

**Review by YutaroHayakawa — APPROVED** [review]

File: cilium-dbg/cmd/bpf_ipcache_update.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+	bpfIPCacheUpdateCmd.PersistentFlags().String("ip", "", "IP address, e.g. 10.244.1.217/32")
+	bpfIPCacheUpdateCmd.PersistentFlags().String("tunnelendpoint", "", "Tunnel endpoint, e.g. 172.21.0.3")
+	bpfIPCacheUpdateCmd.PersistentFlags().Uint32("identity", 0, "Identity, e.g. 14170")
+	bpfIPCacheUpdateCmd.PersistentFlags().Uint8("encryptkey", 0, "Encrypt key")
+	bpfIPCacheUpdateCmd.PersistentFlags().Bool("skiptunnel", false, "Skip tunnel")
+}
+
+var bpfIPCacheUpdateCmd = &cobra.Command{
+	Use:   "update",
+	Short: "Update an entry for an ip",
```
Comment (squeed) [inline cilium-dbg/cmd/bpf_ipcache_update.go]: suggestion: in most cases, this should be called a prefix, not an IP, as the ipcache accepts arbitrary length prefixes.

File: cilium-dbg/cmd/bpf_ipcache_update.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+		skipTunnel, err := cmd.Flags().GetBool("skiptunnel")
+		if err != nil {
+			Usagef(cmd, "Invalid skip tunnel. "+usage)
+		}
+		var flags ipcache.RemoteEndpointInfoFlags
+		if skipTunnel {
+			flags |= ipcache.FlagSkipTunnel
+		}
+
+		key := ipcache.NewKey(ipnet.IP, ipnet.Mask, 0)
```
Comment (squeed) [inline cilium-dbg/cmd/bpf_ipcache_update.go]: Cluster-id should be a parameter as well, if we're going to allow arbitrary low-level access.

File: cilium-dbg/cmd/bpf_ipcache_update.go [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+		}
+
+		tunnelEndpointString, err := cmd.Flags().GetString("tunnelendpoint")
+		if err != nil {
+			Usagef(cmd, "Invalid tunnel endpoint. "+usage)
+		}
+		tunnelEndpoint := net.ParseIP(tunnelEndpointString)
+		if tunnelEndpoint == nil {
+			Usagef(cmd, "Invalid tunnel endpoint. "+usage)
+		}
```
Comment (squeed) [inline cilium-dbg/cmd/bpf_ipcache_update.go:52]: Should reject ipv6 tunnel endpoints here.

File: cilium-dbg/cmd/bpf_ipcache_delete.go [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+var bpfIPCacheDeleteCmd = &cobra.Command{
+	Use:   "delete",
+	Short: "Delete an entry for an ip",
+	Run: func(cmd *cobra.Command, args []string) {
+		common.RequireRootPrivilege("cilium bpf ipcache delete")
+
+		if len(args) < 1 || args[0] == "" {
+			Usagef(cmd, "No ip provided. "+usage)
```
Comment (squeed) [inline cilium-dbg/cmd/bpf_ipcache_delete.go]: nit: prefix, not ip

File: cilium-dbg/cmd/bpf_ipcache_delete.go [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+		}
+
+		arg := args[0]
+
+		ip := net.ParseIP(arg)
```
Comment (squeed) [inline cilium-dbg/cmd/bpf_ipcache_delete.go]: `ParsePrefix()`, not `ParseIP()`

**Review by squeed — CHANGES_REQUESTED** [review]
Basic idea is fine, just some cleanups.

Do we need to add any warnings that this is a very low-level tool? Or is it sufficiently hidden that nobody will find it (and break their cluster)?

**Review by squeed — APPROVED** [review]

**Pull request merged**